src/services/admin.rs

```diff
@@ -551,3 +541,339 @@ impl AdminService {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::node::LightningNode;
+    use crate::services::PaginationRequest;
+    use bitcoin::{Address, Amount, Network};
+    use bitcoincore_rpc::RpcApi;
+    use bitcoind::BitcoinD;
```

Comment:
Since this is a system test that has external dependencies, it should probably not try to masquerade as a unit test.  Those should run quickly and not have external dependencies.

It could be inserted as an integration test, but even that gets run by `cargo test`, which again, should not have external dependencies.

I would have system tests be organized a separate binary in the `Cargo.toml` file and/or as a separate crate.

To do this, you might need to make the relevant parts of the Rust API `pub`.

Reply:
Hm. I'm not saying it makes it 'correct' but this is also how bdk writes their tests.

It's not _exactly_ an external dependency.  The whole point of the bitcoind crate is to avoid you having to setup an externally running bitcoind for your tests.

Given that we have zero unit tests it's probably fine to leave this for now but would make sense to do once we have them?

Reply:
It's my impression that it still requires installing the bitcoind binary and starts a separate process.  Also, it sounds like `cargo check --all-targets --profile=test` would fail for projects where bdk is a dependency unless bitcoind is in the path, which is not what one would expect.

In any case, I don't have a strong opinion about the path there - just a general comment.